src/object/srv_mod.c

```diff
@@ -60,36 +60,96 @@ obj_mod_fini(void)
 /* Define for cont_rpcs[] array population below.
  * See OBJ_PROTO_*_RPC_LIST macro definition
  */
-#define X(a, b, c, d, e)	\
+#define X(a, b, c, d, e, f)	\
 {				\
 	.dr_opc       = a,	\
 	.dr_hdlr      = d,	\
 	.dr_corpc_ops = e,	\
-}
+},
 
 static struct daos_rpc_handler obj_handlers[] = {
-	OBJ_PROTO_CLI_RPC_LIST,
+	OBJ_PROTO_CLI_RPC_LIST
 };
 
 #undef X
 
 static void *
-obj_tls_init(const struct dss_thread_local_storage *dtls,
-	     struct dss_module_key *key)
+obj_tls_init(int xs_id, int tgt_id)
 {
-	struct obj_tls *tls;
+	struct obj_tls	*tls;
+	uint32_t	opc;
+	char		*path;
+	int		rc;
 
 	D_ALLOC_PTR(tls);
 	if (tls == NULL)
 		return NULL;
 
 	D_INIT_LIST_HEAD(&tls->ot_pool_list);
+
+	if (tgt_id < 0)
+		/** skip sensor setup on system xstreams */
+		return tls;
+
+	/** register different per-opcode sensors */
+	for (opc = 0; opc < OBJ_PROTO_CLI_COUNT; opc++) {
+		/** Start with latency, of type duration */
+		D_ASPRINTF(path, "io/%u/ops/%s/latency_us", tgt_id,
+			   obj_opc_to_str(opc));
+		rc = d_tm_add_metric(&tls->ot_op_lat[opc], path,
+				     D_TM_DURATION | D_TM_CLOCK_REALTIME, "",
+				     "");
```

Comment:
FWIW, it would be nice to use the short/long desc fields for extra information. The Go consumer and Prometheus integration can use these to provide additional context for a given metric.